### PR TITLE
Build and Docker workflow refactor

### PR DESCRIPTION
**Pull request by diogormendes**
###### Before submitting the PR

We were building and sending an image to the docker hub on every push.

###### After submitting the PR

Now we have 2 different workflows:

- `uberjar.yml` --> Will build on every push to `master` or `release` branches and will also run the `Trivy` security scan

Images will be sent to `metabase/metabase-enterprise-head:latest` and `metabase/metabase-head:latest` depending on the branch (master or release)

- `dockerhub-branch.yml` --> Will build when there is a comment `@metabase-bot create docker` on the PR. This workflow does not run the `Trivy` security scan

Images will be sent to `metabase/metabase-dev:branch`

### CODE REVIEW

**Comment by diogormendes** [comment]
@metabase-bot create docker

**Comment by diogormendes** [comment]
@metabase-bot create docker

File: .github/workflows/dockerhub-branch.yml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    paths-ignore:
+    - 'docs/**'
+    - 'frontend/test/**'
+    - 'enterprise/frontend/test/**'
+    - ".**"
+    - "test*"
+
+jobs:
+  build:
+    if: github.event.comment.body == '@metabase-bot create docker'
```
Comment (paoliniluis) [inline .github/workflows/dockerhub-branch.yml:16]: create branch image?

**Review by dragonsahead — APPROVED** [review]
LGTM, I agree with @ariya that we should test this first in a fork

**Comment by diogormendes** [comment]
Comments are working as expected on my fork ([link](https://github.com/diogormendes/metabase_fork/actions/workflows/dockerhub-branch.yml)), but for some reason, the merge to master is not, even though I'm pretty sure the config is right. Some ideas here would be great @paoliniluis @ariya .

I tough about adding `pull_request: [closed]` as well

**Comment by nemanjaglumac** [comment]
Closing as it became obsolete after #26680

**Pull request closed without merging**